docs/trace_collection.md

```diff
@@ -279,55 +522,126 @@ To provide a continuous trace inside a RxJava stream you need to follow the step
    ```
 2. Then in your project open a scope when the Observable is subscribed and close it when it completes. Any span
    created inside the stream operators will be displayed inside this scope (parent Span):
-   ```kotlin
-   Single.fromSupplier{}
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+Single.fromSupplier{} 
         .subscribeOn(Schedulers.io())
-        .map { 
+        .map {
             val span = GlobalTracer.get().buildSpan("<YOUR_OP_NAME>").start()
             // ...
             span.finish()
         }
-       .doOnSubscribe {
-           val span = GlobalTracer.get()
-               .buildSpan("<YOUR_OP_NAME>")
-               .start()
-           GlobalTracer.get().scopeManager().activate(span)
-       }
-       .doFinally {
-           GlobalTracer.get().scopeManager().activeSpan()?.let {
-               it.finish()
+        .doOnSubscribe {
+            val span = GlobalTracer.get()
+                    .buildSpan("<YOUR_OP_NAME>")
+                    .start()
+            GlobalTracer.get().scopeManager().activate(span)
+        }
+        .doFinally {
+            GlobalTracer.get().scopeManager().activeSpan()?.let {
+                it.finish()
+            }
+        }
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+Single.fromSupplier({})
+        .subscribeOn(Schedulers.io())
+        .map { object -> {
+            final Span span = GlobalTracer.get().buildSpan("<YOUR_OP_NAME>").start();
+            // ...
+            span.finish();
+            // ...
            }
-       }
-   ```
+        )
+        .doOnSubscribe {
+            final Span span = GlobalTracer.get().buildSpan("<YOUR_OP_NAME>").start();
+            GlobalTracer.get().scopeManager().activate(span);
+        }
+        .doFinally {
+            final Span activeSpan = GlobalTracer.get().scopeManager().activeSpan();
+            if (activeSpan != null) {
+                activeSpan.finish();
+            }
+        }
```

Comment:
Syntax for Java will be different from Kotlin in case of lambda usage. With Java we still need to provide parenthesis even if the argument is lambda.